pkg/inst-api-semconv/instrumenter/http/http_status_code_extractor_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"testing"
+)
+
+type testSpan struct {
+	trace.Span
+	status *codes.Code
+	Kvs    []attribute.KeyValue
+}
+
+func (ts *testSpan) SetStatus(status codes.Code, desc string) {
+	*ts.status = status
+}
+
+func (ts *testSpan) SetAttributes(kv ...attribute.KeyValue) {
+	ts.Kvs = kv
+}
+
+type testReadOnlySpan struct {
+	sdktrace.ReadWriteSpan
+	isRecording bool
+}
+
+func (t *testReadOnlySpan) Name() string {
+	return "http-route"
+}
+
+func (t *testReadOnlySpan) IsRecording() bool {
+	return t.isRecording
+}
+
+type customizedNetHttpAttrsGetter struct {
+	code int
+}
+
+func (c customizedNetHttpAttrsGetter) GetRequestMethod(request any) string {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c customizedNetHttpAttrsGetter) GetHttpRequestHeader(request any, name string) []string {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c customizedNetHttpAttrsGetter) GetHttpResponseStatusCode(request any, response any, err error) int {
+	return c.code
+}
+
+func (c customizedNetHttpAttrsGetter) GetHttpResponseHeader(request any, response any, name string) []string {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c customizedNetHttpAttrsGetter) GetErrorType(request any, response any, err error) string {
+	//TODO implement me
+	panic("implement me")
+}
+
+func TestHttpClientSpanStatusExtractor500(t *testing.T) {
+	c := HttpClientSpanStatusExtractor[any, any]{
+		Getter: customizedNetHttpAttrsGetter{
+			code: 500,
+		},
+	}
+	u := codes.Code(0)
+	span := &testSpan{status: &u}
+	c.Extract(span, nil, nil, nil)
+	if *span.status != codes.Error {
+		panic("span status should be error!")
+	}
+}
+
+func TestHttpClientSpanStatusExtractor400(t *testing.T) {
+	c := HttpClientSpanStatusExtractor[any, any]{
+		Getter: customizedNetHttpAttrsGetter{
+			code: 400,
+		},
+	}
+	u := codes.Code(0)
+	span := &testSpan{status: &u}
+	c.Extract(span, nil, nil, nil)
+	if *span.status != codes.Error {
+		panic("span status should be error!")
+	}
+	if span.Kvs == nil {
```

Comment:
It looks like this line is testing that we set the span attribute `error.type=<value>`. Can we change the assert to specifically test against the `error.type` key?